src/main/kotlin/mobi/waterdog/eventbus/EventBusProvider.kt

```diff
@@ -0,0 +1,41 @@
+package mobi.waterdog.eventbus
+
+import mobi.waterdog.eventbus.engine.EventProvider
+import mobi.waterdog.eventbus.engine.kafka.KafkaEventProvider
+import mobi.waterdog.eventbus.engine.local.LocalEventProvider
+import mobi.waterdog.eventbus.persistence.LocalEventStore
+import java.util.Properties
+
+class EventBusProvider(type: EventBackend) {
+
+    companion object {
+        const val CLEANUP_INTERVAL_SECONDS_PROP = "producer.event.cleanup.intervalInSeconds"
+    }
+
+    private val provider: EventProvider = when (type) {
+        EventBackend.Local -> LocalEventProvider()
+        EventBackend.Kafka -> KafkaEventProvider()
+    }
+
+    private var localEventStore: LocalEventStore? = null
+
+    fun setupProducer(localEventStore: LocalEventStore) {
```

Comment:
I have this separate setupProducer to deal with the local event store because consumers don't need that.